woocommerce/woocommerce
# CYS - Core: ensure that the the default font pair and color pair are selected

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR ensures that the first font pair and first color pair that are visible in the assembler sidebar are applied in the preview and highlighted in the Assembler sidebar.

Closes #44258.
Closes #44307.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
3. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
4. Install this build of Gutenberg that contains the new Font Library API: 
[gutenberg.zip](https://github.com/woocommerce/woocommerce/files/14085546/gutenberg.zip)
5. Visit the `wp-admin/admin.php?page=wc-admin&path=/customize-store`.
6. Follow the process.
7. Click on "Change Fonts"
8. Ensure that the first font pair is highlighted.
9. Go back.
10. Click on "Change Color Palette".
11. Ensure that the first color pair is highlighted.


https://github.com/woocommerce/woocommerce/assets/4463174/c6608236-b3a0-480c-85ae-f0c73dd7a091



Given that these changes impact some code shared with the AI flow, please ensure that the AI flow isn't impacted:

### JN Installation


1. Create a new WooCommerce installation with this version.
2. Ensure Jetpack is installed and your site is connected with JP.
3. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
4. Ensure the Woo AI plugin is installed and activated (available on this monorepo).
5. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
6. Head over to WooCommerce -> Home and click on "Start Customizing".
7. Ensure that all the flow works correctly.






### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS - Core: ensure that the the default font pair and color pair are selected.

#### Comment 

</details>
